pkg/operator/reports.go

```diff
@@ -859,6 +924,58 @@ func (op *Reporting) queueDependentReportsForReport(report *metering.Report) err
 	return nil
 }
 
+func (op *Reporting) doReportsDependOnThisReport(report *metering.Report) bool {
+	// Look for all reports in the namespace
+	reports, err := op.reportLister.Reports(report.Namespace).List(labels.Everything())
+	if err != nil {
+		log.Errorf("Was unable to list reports for report: %s, while looking for dependent reports")
+		return false
+	}
+
+	// for each report in the namespace, find ones that depend on the report passed into the function.
+	for _, otherReport := range reports {
+		deps, err := op.getReportDependencies(otherReport)
+		if err != nil {
+			log.Errorf("Was unable to get dependencies for report: %s, while looking for dependent reports")
+			return false
+		}
+		// If a Report has a dependency on the passed in report, we're done
+		for _, dep := range deps.Reports {
+			if dep.Name == report.Name {
+				break
+			}
+		}
+	}
+	// true means no other reports depend on this report
+	return true
+}
+
+func (op *Reporting) doReportQueriesDependOnThisReport(reportQuery *metering.ReportQuery) bool {
+	// Look for all ReportQueries in the namespace
+	reportQueries, err := op.reportQueryLister.ReportQueries(reportQuery.Namespace).List(labels.Everything())
+	if err != nil {
+		log.Errorf("Was unable to list reports for report: %s, while looking for dependent queries")
```

Comment:
A couple of things here:
- Probably want to use the initialized `op.Logger` (or whatever the name of that field may be).
- You specified a format specification but didn't provide the corresponding variable.

Reply:
I was modeling this off of the func and usage for `isReportFinished()` which seems to pass around the reference to `logger` that `runReport()` has itself been passed. Is this a bad model to follow?

Reply:
I was modeling this off of the func and usage for `isReportFinished()` which seems to pass around the reference to `logger` that `runReport()` has itself been passed. Is this a bad model to follow?

Reply:
however just changed them all to op.logger, which does seem to log properly, so...